File: src/converter/plugins/convertPropertyAccessExpression.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {CheckCoverageService, checkCoverageServiceKey} from "./CheckCoveragePlugin.js";

export const convertPropertyAccessExpression = createSimplePlugin((node, context, render) => {
    if (!ts.isPropertyAccessExpression(node)) return null

    const checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    return `${render(node.expression)}.${render(node.name)}`
})

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertPropertyAccessExpression.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createSimplePlugin
import typescript.Node
import typescript.isPropertyAccessExpression

val convertPropertyAccessExpression = createSimplePlugin plugin@{ node: Node, context, render ->
    if (!isPropertyAccessExpression(node)) return@plugin null

    val checkCoverageService = context.lookupService<CheckCoverageService>(checkCoverageServiceKey)
    checkCoverageService?.cover(node)

    "${render(node.expression)}.${render(node.name)}"
}
